modules/spring-config/src/main/java/org/mule/runtime/config/api/LazyComponentInitializer.java

```diff
@@ -49,6 +49,22 @@ public interface LazyComponentInitializer {
    */
   void initializeComponent(Location location);
 
+  /**
+   * Calling this method guarantees that the requested component from the configuration and its dependencies will be created,
+   * initialized and started.
+   * <p/>
+   * The requested component must exists in the configuration.
+   * <p/>
+   * Unlike {@link #initializeComponent(Location)} and {@link #initializeComponents(ComponentLocationFilter)}, calling this method
+   * will not affect previously created/started components.
+   *
+   * @param location the location of the configuration component.
+   * @throws MuleRuntimeException if there's a problem creating the component or the component does not exist.
+   *
+   * @since 4.6
+   */
+  void initializeAdditionalComponent(Location location);
```

Comment:
Let me know if you believe there is no much benefit from exposing this method as API.
This is not needed for the original intent because we are doing it transparently when there is a registry lookup miss.
But it may be useful to have manual control to speed up initialization in cases where the components are known beforehand and need to be enabled incrementally, in stages. See **Extended Scope II** in the PR [description](https://github.com/mulesoft/mule/pull/12183#issue-1553479348).

Reply:
Agreed, will do.
I just will have to update my tests a little bit because they use this method, but I should be able to achieve something similar by calling the lookup on the registry.
I will let you know when it's done.

Reply:
Done, this method is now private on `LazyMuleArtifactContext` and has been removed from the `LazyComponentInitializer` interface.